blog/en/blog/2022/09/09/kubernetes-ingress-with-apisix.md

```diff
@@ -0,0 +1,248 @@
+
+To configure Routes, APISIX comes with declarative and easy-to-use [custom resource](https://apisix.apache.org/docs/ingress-controller/next/references/apisix_route_v2beta3/):
+
+```yaml {title="apisix-ingress-manifest.yaml"}
+apiVersion: apisix.apache.org/v2beta2
+kind: ApisixRoute
+metadata:
+  name: api-routes
+spec:
+  http:
+    - name: route-1
+      match:
+        hosts:
+          - local.navendu.me
+        paths:
+          - /v1
+      backends:
+        - serviceName: bare-minimum-api-v1
+          servicePort: 8080
+    - name: route-2
+      match:
+        hosts:
+          - local.navendu.me
+        paths:
+          - /v2
+      backends:
+        - serviceName: bare-minimum-api-v2
+          servicePort: 8081
+```
+
+The APISIX Ingress controller converts this resource to an APISIX gateway configuration.
+
+APISIX also supports configuration using native [Kubernetes Ingress resource](https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource):
+
+```yaml {title="kubernetes-ingress-manifest.yaml"}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: api-routes
+spec:
+  ingressClassName: apisix
+  rules:
+    - host: local.navendu.me
+      http:
+        paths:
+          - backend:
+              service:
+                name: bare-minimum-api-v1
+                port:
+                  number: 8080
+            path: /v1
+            pathType: Exact
+          - backend:
+              service:
+                name: bare-minimum-api-v2
+                port:
+                  number: 8081
+            path: /v2
+            pathType: Exact
+```
+
+You can use either to configure APISIX but I prefer the easier APISIX custom resource. We can apply this manifest file to our cluster to create Routes in APISIX:
+
+```shell
+kubectl apply -f apisix-ingress-manifest.yaml
+```
+
+If the Ingress controller is configured correctly, you should see a response indicating that APISIX API gateway has been configured:
+
+```shell {title="output"}
+apisixroute.apisix.apache.org/api-routes created
+```
+
+Now, let's test these Routes.
+
+## Testing the Created Routes
+
+If you were following along using minikube and `NodePort`, you should be able to access APISIX through the Node IP of the service `apisix-gateway`. If the Node IP is not reachable directly (if you are on Darwin, Windows, or WSL), you can create a tunnel to access the service on your machine:
+
+```shell
+minikube service apisix-gateway --url -n ingress-apisix
+```
+
+This will show the URL with which you can access the `apisix-gateway` service.
+
+You can send a `GET` request to this URL and it would be Routed to the appropriate service:
+
+```shell
+curl http://127.0.0.1:51538/v2 -H 'host:local.navendu.me'
```

Comment:
Maybe add a `/v1` access example?

Reply:
I have mentioned v1 route in the example. It might be unnecessary repetition.